vertical-pod-autoscaler/pkg/utils/vpa/sequential_processor.go

```diff
@@ -31,19 +33,28 @@ type sequentialRecommendationProcessor struct {
 }
 
 // Apply chains calls to underlying RecommendationProcessors in order provided on object construction
-func (p *sequentialRecommendationProcessor) Apply(podRecommendation *vpa_types.RecommendedPodResources,
-	policy *vpa_types.PodResourcePolicy,
-	conditions []vpa_types.VerticalPodAutoscalerCondition,
+func (p *sequentialRecommendationProcessor) Apply(
+	vpa *vpa_types.VerticalPodAutoscaler,
 	pod *v1.Pod) (*vpa_types.RecommendedPodResources, ContainerToAnnotationsMap, error) {
-	recommendation := podRecommendation
+
+	if vpa == nil {
+		return nil, nil, fmt.Errorf("cannot process nil vpa")
+	}
+	if vpa.Status.Recommendation == nil {
+		return nil, nil, nil
+	}
+
+	var recommendation *vpa_types.RecommendedPodResources
```

Comment:
One small edge case here is that if there are no processors, this will be nil (where the behavior beforehand was that we would return the existing VPA recommendation).

I know we don't currently have an empty processor but should we match the behavior here?

Reply:
🤦 Yup, good callout.
Should be fixed in 59236c93a2